src/nvim/terminal.c

```diff
@@ -217,10 +216,14 @@ Terminal *terminal_open(TerminalOptions opts)
   vterm_set_utf8(rv->vt, 1);
   // Setup state
   VTermState *state = vterm_obtain_state(rv->vt);
+  // Only relevant in terminal true color mode
+  if (true_color) {
```

Comment:
I know we don't like adding new configuration vars, but instead of enabling by default if true color is enabled, personally, I'd prefer a new option like 'g:terminal_color_[0-15]' to control this behaviour.